### ISSUE TITLE
VMware ESXi Not Detecting Guest OS is CentOS 7

### ISSUE THREAD

**Issue by pdugas** [vmware/open-vm-tools#135]
Apologies in advance if this isn't the right place to ask this but I have a few CentOS 7 64-bit VMs running on a single ESXi 6.5 host.  The VMs were built from the minimal ISO then yum-updated and then open-vm-tools was installed from the stock repositories (version 10.0.5).  The ESXi web interface is warning that the tools aren't reporting the same version of the OS that I selected when I built them.

> The configured guest OS (CentOS 7 (64-bit)) for this virtual machine does not match the guest that is currently running (CentOS 4/5 or later (64-bit)). You should specify the correct guest OS to allow for guest-specific optimizations.

Is there a way to adjust open-vm-tools to report the correct value?  

```
[root@www ~]# ls -l /etc/*-release
-rw-r--r--. 1 root root  38 Nov 29 13:12 /etc/centos-release
-rw-r--r--. 1 root root 393 Nov 29 13:12 /etc/os-release
lrwxrwxrwx. 1 root root  14 Feb  7 19:59 /etc/redhat-release -> centos-release
lrwxrwxrwx. 1 root root  14 Feb  7 19:59 /etc/system-release -> centos-release
[root@www ~]# cat /etc/centos-release
CentOS Linux release 7.3.1611 (Core) 
[root@www ~]# cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

[root@www ~]# 
```

**Comment by lemke1458** [comment]
This is fixed in 10.1.0

The 10.0.5 code doesn't try to differentiate CentOS versions:

https://github.com/vmware/open-vm-tools/blob/stable-10.0.x/open-vm-tools/lib/misc/hostinfoPosix.c#L582

10.1.x does
https://github.com/vmware/open-vm-tools/blob/stable-10.1.x/open-vm-tools/lib/misc/hostinfoPosix.c#L585

**Comment by pdugas** [comment]
Roger.  Thanks much.
